consensus/src/main/kotlin/maru/consensus/ProposerSelector.kt

```diff
@@ -0,0 +1,24 @@
+/*
+   Copyright 2025 Consensys Software Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.consensus
+
+import maru.core.BeaconBlockHeader
+import maru.core.Validator
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+
+interface ProposerSelector {
+  fun getProposerForBlock(header: BeaconBlockHeader): SafeFuture<Validator>
```

Comment:
Jason mentioned `selectProposerForRound` which has `ConsensusRoundIdentifier` as an argument. So I believe `BeaconBlockHeader` won't be enough and we should also use `ConsensusRoundIdentifier` here.

Also, I believe `ProposerSelector` will be a QBFT specific thing, so we can move it under `maru.consensus.qbft` package.

Reply:
`ProposerSelector` is definitely QBFT specific thing so belongs under qbft package. We don't want the block header as an argument for the `ProposerSelector` just the round and height  or `ConsensusRoundIdentifier`. We can use the static method https://github.com/hyperledger/besu/blob/main/consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/blockcreation/ProposerSelector.java#L115 from `ProposerSelector` look up the parent block for the validators and the previous proposer.

Reply:
A proposer is something defined by protocol rules, so the way to select it doesn't necessarily defined solely by parent's block header. At least the way `BeaconBlockBody` is. `BeaconBlockHeader` could define the proposer only if we could produce a block without an execution payload, only with a state root change. And the state transition would consist only of signed QBFT messages proving that consensus wasn't reached on a given round